zzk/service/assignment.go

```diff
@@ -136,48 +132,102 @@ func (h *ZKAssignmentHandler) getAssignedHostID(poolID, ipAddress string) (strin
 	return "", ErrNoAssignedHost
 }
 
+func getIDString(hosts []host.Host) string {
+	hostIDs := []string{}
+
+	for _, h := range hosts {
+		hostIDs = append(hostIDs, h.ID+":"+h.PoolID)
+	}
+
+	return strings.Join(hostIDs, ",")
+}
+
 func (h *ZKAssignmentHandler) assignToHost(poolID, ipAddress, netmask, binding string, cancel <-chan interface{}) error {
```

Comment:
That is intentional.  In order to fix and clean up the dependencies, the pool listener no longer uses the host registry listener to look for hosts.  It used to block until a host was online, but now it polls every 5s for a host so the channel is no longer needed.